task-generator/remote/main.go

```diff
@@ -160,9 +187,19 @@ fi
 		script := "scripts/script-" + step.Name + ".sh"
 
 		ret += "\ncat >" + script + " <<'REMOTESSHEOF'\n"
-		if !strings.HasPrefix(step.Script, "#!") {
+
+		// The base task might now be using a bash shell, so we need to make sure
+		// that we only have one shebang declaration. If there is a shebang declaration,
+		// we should also consolidate the set declarations.
+		reShebang := regexp.MustCompile(`(#!.*\n)(set -.*\n)*`)
```

Comment:
This is for the script that ends up getting executed on the VM, right? I don't think we should mess with its shebang :thinking: 

Only if it doesn't have one at all, in which case we should prepend the same thing that Tekton would
```
#!/bin/sh
set -e
```

Reply:
Without this change, the `cd` operation is in the wrong location:
```
       cat >scripts/script-build.sh <<'REMOTESSHEOF'
+      cd /var/workdir
       #!/bin/bash
       set -e
-      cd /var/workdir
```

Reply:
This is because we are adding the shebang to the beginning of the script in the base task so we can no longer simply concatenate it after setting the workingDir.

Reply:
Ah, right